### ISSUE TITLE
[Twig Bridge] form attribute is not inherited by children for expanded choice widgets

### ISSUE THREAD

**Issue by vincentpazeller** [symfony/symfony#33008]
### General issue:

Assume an expanded choice field:
`$builder->add('field', ChoiceType::class, ['choices' => ... , 'expanded' => true)`

And rendering it like this:
`{{ form_widget(form.field, {attr: { form: 'my-form' }}) }}`

Then the form attribute is not rendered for child checkboxes/radios...

I think the form attribute should be inherited in that case.

I use the Bootstrap theme (Symfony 3.4), but I think this is a more general issue.

### My workaround
I have a quite complex page with multiple forms (which made this issue show up) and I added this in my "custom form" template file (there is probably a better way to achieve this...):

<pre>
{% block choice_widget_expanded -%}
    <div {{ block('widget_container_attributes') }}>
        {% set inheritedAttrs = {} %}
        {% if attr.form is defined and attr.form is not empty %}
            {% set inheritedAttrs = {form: attr.form|default()} %}
        {% endif %}
        {%- for child in form %}
            {{- form_widget(child, {
                parent_label_class: label_attr.class|default(''),
                attr: inheritedAttrs,
                translation_domain: choice_translation_domain,
                valid: valid,
            }) -}}
        {% endfor -%}
    </div>
{%- endblock choice_widget_expanded %}
</pre>

**Comment by rjwebdev** [comment]
Are you aware of the `choice_attr` option that can be defined in your form? See https://symfony.com/doc/3.4/reference/forms/types/choice.html#choice-attr

**Comment by vincentpazeller** [comment]
Yes, I am aware of it. I think it does not work to write:

`{{form_widget(form.field, {choice_attr: { form: 'my-form' } })`

I think the choice_attr is mainly used in the type definition or controller, not in twig templates...

Anyway, I think that in the case of the form attribute it could be automatically inherited, without further configuration.

**Comment by xabbuh** [comment]
I don't think this is the way to go. Cascading attributes down will likely break lots of forms (especially when there is several nesting of levels). The solution here is indeed to make use of the `choice_attr` option in the form type or tweaking the rendering by overriding some block of the form theme.

**Comment by vincentpazeller** [comment]
If I set the form attribute on a single (expanded) Choice form type, as a programmer I see it as one widget (not an embedded type) and I don't think this is convenient that I need to set it on every single option just because I changed the "expanded" option. I also understand that this is a breaking change but this might be improved in a future breaking version. After all the role of a framework is to make developers' life easier. I think that usually when you set the form attribute on a ChoiceType, if it happens to be expanded, the general case is that you want options to be in that form, so people doing weird things should have to write extra code, not people doing standard things... Also, has anyone an example of using the choice_attr in a twig template? I saw examples within controllers, but could not make it work in a twig template... Thanks
